src/bp/core/services/telemetry/legacy-stats.ts

```diff
@@ -0,0 +1,187 @@
+import LicensingService from 'common/licensing-service'
+import { machineUUID } from 'common/stats'
+import { ConfigProvider } from 'core/config/config-loader'
+import Database from 'core/database'
+import { UserRepository } from 'core/repositories'
+import { TelemetryRepository } from 'core/repositories/telemetry_payload'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import ms from 'ms'
+import os from 'os'
+import uuid from 'uuid'
+
+import { GhostService } from '..'
+import AuthService from '../auth/auth-service'
+import { BotService } from '../bot-service'
+import { CMSService } from '../cms'
+import { JobService } from '../job-service'
+import { WorkspaceService } from '../workspace-service'
+
+import { TelemetryStats } from './telemetry-stats'
+
+@injectable()
+export class LegacyStats extends TelemetryStats {
+  protected url: string
+  protected lock: string
+  protected interval: number
+
+  constructor(
+    @inject(TYPES.GhostService) ghostService: GhostService,
+    @inject(TYPES.Database) database: Database,
+    @inject(TYPES.LicensingService) licenseService: LicensingService,
+    @inject(TYPES.JobService) jobService: JobService,
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository,
+    @inject(TYPES.ConfigProvider) private config: ConfigProvider,
+    @inject(TYPES.CMSService) private cmsService: CMSService,
+    @inject(TYPES.AuthService) private authService: AuthService,
+    @inject(TYPES.UserRepository) private userRepository: UserRepository,
+    @inject(TYPES.BotService) private botService: BotService,
+    @inject(TYPES.WorkspaceService) private workspaceService: WorkspaceService
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = 'https://telemetry.botpress.io/ingest'
+    this.lock = 'botpress:telemetry-legacyStats'
+    this.interval = ms('1d')
```

Comment:
These variables could be declared in the parent class instead of duplicating them in child classes

Reply:
I need them in the child classes since they are not the same every time

Reply:
URL and interval seems to be the same in both classes, just lock is changing

Reply:
'https://telemetry.botpress.io/ingest' vs 'https://telemetry.botpress.cloud/ingest'